File: codes/golang/621.任务调度器.go

func leastInterval(tasks []byte, n int) int {
	if len(tasks) == 0 {
		return 0
	}
	size := len(tasks)
	wordCnt := make([]int, 26)
	for _, task := range tasks {
		index := task - 'A'
		wordCnt[index] += 1
	}
	maxCnt := 1
	for _, cnt := range wordCnt {
		if cnt > maxCnt {
			maxCnt = cnt
		}
	}
	tot := 0
	for _, cnt := range wordCnt {
		if cnt == maxCnt {
			tot += 1
		}
	}
	return max(size, (maxCnt-1)*(n+1)+tot)
}

File: codes/python/621.任务调度器.py

class Solution:
	def leastInterval(self, tasks: List[str], n: int) -> int:
		if len(tasks) == 0:
			return 0
		size = len(tasks)
		wordCnt = [0] * 26
		for task in tasks:
			index = ord(task) - ord('A')
			wordCnt[index] += 1
		maxCnt = 1
		for cnt in wordCnt:
			if cnt > maxCnt:
				maxCnt = cnt
		tot = 0
		for cnt in wordCnt:
			if cnt == maxCnt:
				tot += 1
		return max(size, (maxCnt-1)*(n+1)+tot)